### ISSUE TITLE
1. YOLO26n inference latency much higher than official benchmark on Jetson AGX Orin

### ISSUE THREAD

**Issue by 323-w** [ultralytics/ultralytics#25030]
### Search before asking

- [x] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/orgs/ultralytics/discussions) and found no similar questions.

### Question

I noticed from your official website that inference using the engine file takes approximately 5 ms per image. I’ve set my device to maximum power mode for testing.
 
I first benchmarked inference speed with the PT file by running the command:
 yolo predict model=yolo26n.pt source=./imgs imgsz=640 
 
To eliminate frame rate degradation caused by model warm-up, I copied 100 official test images into the  ./imgs  folder for the test. However, the resulting inference speed I measured is around 25 ms per image, while the official documentation states the latency is under 10 ms per image. Could you please advise what steps I might have done incorrectly?

**Comment by UltralyticsAssistant** [comment]
👋 Hello @323-w, thank you for your interest in Ultralytics 🚀! This is an automated response, and an Ultralytics engineer will also assist soon. We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can find many [Python](https://docs.ultralytics.com/usage/python) and [CLI](https://docs.ultralytics.com/usage/cli) usage examples and where many of the most common questions may already be answered.

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example) to help us debug it.

For inference latency and benchmark ❓ Questions, please provide as much information as possible, including the exact export command, inference command, device details, JetPack/CUDA/CUDNN/PyTorch versions, `ultralytics` version, logs, and whether you are testing a `.pt` or exported engine file. Please also verify you are following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips) where applicable.

Join the Ultralytics community where it suits you best. For real-time chat, head to [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer in-depth discussions? Check out [Discourse](https://community.ultralytics.com/). Or dive into threads on our [Subreddit](https://www.reddit.com/r/Ultralytics/) to share knowledge with the community.

## Upgrade

Upgrade to the latest `ultralytics` package including all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) to verify your issue is not already resolved in the latest version:

```bash
pip install -U ultralytics
```

## Environments

YOLO may be run in any of the following up-to-date verified environments (with all dependencies including [CUDA](https://developer.nvidia.com/cuda)/[CUDNN](https://developer.nvidia.com/cudnn), [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/) preinstalled):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo26"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM. See [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial)
- **Amazon** Deep Learning AMI. See [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial)
- **Docker Image**. See [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests verify correct operation of all YOLO [Modes](https://docs.ultralytics.com/modes) and [Tasks](https://docs.ultralytics.com/tasks) on macOS, Windows, and Ubuntu every 24 hours and on every commit.

**Comment by Y-T-G** [comment]
Hello.

Can you post the latency logs from the inference?

**Comment by 323-w** [comment]
Thank you for your reply. I’ve resolved the issue now. I just wanted to test the engine file. After converting the model to engine format, I found its inference speed matches the official benchmark perfectly.

**Comment by 323-w** [comment]
Sorry to bother you again.

<img width="1611" height="288" alt="Image" src="https://github.com/user-attachments/assets/35e6176b-2053-4b96-8cb9-4a4d20c7322b" />

<img width="791" height="992" alt="Image" src="https://github.com/user-attachments/assets/157d53ee-ef26-4075-89a9-59be98cfab8d" />

 I’ve run into a new issue. Today I reinstalled the system via SDK Manager, which installed Ubuntu 24 and CUDA 13.2 by default. When I ran inference with the command  yolo predict model=yolo26n.pt imgsz=640 , the model produced erroneous detection results. The log showed that the test images were incorrectly detected as cakes. I downloaded the official weight file directly. Could you tell me what might be causing this problem and how to resolve it

**Comment by Y-T-G** [comment]
@323-w Is CPU inference correct? `device=cpu`

**Comment by 323-w** [comment]
Thank you for your reply. The inference results on CPU are correct, while those on GPU are incorrect. Could you please advise what the potential causes might be?

**Comment by 323-w** [comment]
Could this be an issue with the PyTorch environment? I plan to recreate a virtual environment. Are there any dedicated commands for installing PyTorch on Jetson devices?

**Comment by 323-w** [comment]
Thanks, I’ve fixed it by reinstalling Torch

**Comment by glenn-jocher** [comment]
Glad you got it working. Since CPU results were correct but GPU results were wrong, reinstalling a Jetson-compatible `torch`/CUDA stack was the right fix and points to an environment mismatch rather than a model issue; for future Jetson setup, the [Ultralytics NVIDIA Jetson guide](https://docs.ultralytics.com/guides/nvidia-jetson) is the best reference, and TensorRT/engine export is still recommended for matching the published Jetson latency benchmarks.
